ivy/data_classes/array/array.py

```diff
@@ -1144,10 +1144,10 @@ def __deepcopy__(self, memodict={}):
                 jax_array = ivy.array(np_array)
                 return to_ivy(jax_array)
             return to_ivy(copy.deepcopy(self._data))
-        except RuntimeError:
```

Comment:
I think we can update this `except` block to something like this instead,
```
        except RuntimeError:
            from ivy.functional.ivy.gradients import _is_variable
            if ivy.current_backend_str() == "paddle":
                return to_ivy(copy.deepcopy(self._data.numpy()))
            elif _is_variable(self):
                return to_ivy(copy.deepcopy(ivy.stop_gradient(self)._data))
            return to_ivy(copy.deepcopy(self._data))
```
Could you please also add some commit above the `elif` stating the leaf tensors issue? Thanks!